yarn-project/simulator/src/public/enqueued_calls_processor.ts

```diff
@@ -249,14 +313,16 @@ export class EnqueuedCallsProcessor {
       const availableGas = this.getAvailableGas(tx, publicKernelOutput, phase);
       const transactionFee = this.getTransactionFee(tx, publicKernelOutput, phase);
 
+      // each enqueued call starts with an incremented side effect counter
+      const enqueuedCallStateManager = txStateManager.fork(/*incrementSideEffectCounter=*/ true);
```

Comment:
IDK if I like the `incrementSideEffectCounter` flag. I wonder if we could either
* always increment? (maybe cannot be done now because it would break sync with kernel, circuit etc, but maybe it's ok later)
* do a fork() AND a .incrementSideEffectCounter() when needed

(ok to change in other PR if needed)